Based on pull request: https://github.com/magento/magento2/commit/595a7a8f56571a5bd2753096bef9db27d46020ff#diff-632480c00239b903f76daee7e1733bff
Should fix excluded mode - it is the mode set after selectAll action is triggered. Without fix no selected items were passed to request.

diff --git a/view/base/web/js/grid/massactions.js b/view/base/web/js/grid/massactions.js
index a1884c2..3626f52 100644
--- a/view/base/web/js/grid/massactions.js
+++ b/view/base/web/js/grid/massactions.js
@@ -153,6 +153,11 @@ define([
             var itemsType = data.excludeMode ? 'excluded' : 'selected',
                 selections = {};
 
+            if (itemsType === 'excluded' && data.selected && data.selected.length) {
+                itemsType = 'selected';
+                data[itemsType] = _.difference(data.selected, data.excluded);
+            }
+
             selections[itemsType] = data[itemsType];
 
             if (!selections[itemsType].length) {